src/components/search/autocomplete.js

```diff
@@ -0,0 +1,107 @@
+import React from "react"
+import axios from "axios"
+import Select, { components } from "react-select"
+import Highlighter from "react-highlight-words"
+
+const getLocationOptions = () => axios.get("/content/locations.json")
+const getZones = () => axios.get("/content/zones.json")
+
+const Autocomplete = ({ setIsZoneDialog, setSelected, selected, setZones }) => {
+  const [locationOptions, setLocationOptions] = React.useState([])
+  const [inputChange, setinputChange] = React.useState()
+
+  const cssBtn = {
+    cursor: "pointer",
+    fontSize: "1.8rem",
+    display: "inline-block",
+    padding: "1rem",
+    background: "var(--cl-black)",
+    color: "var(--cl-white)",
+    fontFamily: "var(--ff-title)",
+    border: "1px solid #000000",
+    borderRadius: " 0 4px 4px 0",
+    display: "flex",
+    alignItems: "center",
+  }
+
+  React.useEffect(() => {
+    let ignore = false
+
+    Promise.all([getLocationOptions(), getZones()]).then(results => {
+      if (!ignore) {
+        setLocationOptions(results[0].data)
+        setZones(results[1].data)
+      }
+    })
+    return () => {
+      ignore = true
```

Comment:
สงสัย logic ตรงนี้มากว่าทำไมต้องมี ignore เพราะ behavior ที่ React ชอบ run useEffect หลายทีเหรอ? ignore เป็น local variable ก็เลยงงว่าใส่ไว้ใน cleanup function แล้วมันจะมีผลอะไรกันนะ @heyfirst เซียน react มีความเห็นกับสิ่งนี้บ่ มันดูยังไงๆ

Reply:
มันถูกแล้ว มันคือ correct way to do useEffect (https://beta.reactjs.org/reference/react/useEffect#fetching-data-with-effects) การ return แบบนี้

Reply:
OH MY GOSH TODAY I LEARN
THAT IS WHY I HATE REACT SO BAD ไม่มีท่าที่คลีนกว่านี้แล้วจริงดิ ทำไมตรูต้องมาดัก race condition เองด้วย  🤣